### PR TITLE
Implement commands for navigating over logical sexps

### PR DESCRIPTION
**Pull request by Malabarba**
Affects https://github.com/clojure-emacs/cider/issues/1171

### CODE REVIEW

File: clojure-mode.el [hunk 1]
```diff
@@ -1044,6 +1044,45 @@ Returns a list pair, e.g. (\"defn\" \"abc\") or (\"deftest\" \"some-test\")."
         (list (match-string 1)
               (match-string 2))))))
 
+
+;;; Sexp navigation
+(defun clojure-mode-forward-logical-sexp (&optional n)
```
Comment (bbatsov) [inline clojure-mode.el]: Major mode commands generally don't have `mode` in their names.

File: test/clojure-mode-sexp-test.el [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Code:
+
+(require 'clojure-mode)
+(require 'ert)
+
+(ert-deftest test-sexp ()
+  (with-temp-buffer
+    (insert "^String #macro ^dynamic reverse")
```
Comment (bbatsov) [inline test/clojure-mode-sexp-test.el]: This probably needs a few more tests showing that it won't go back or forward too far (e.g. to a previous/next expression).

**Comment by bbatsov** [comment]
This also needs a changelog entry.

**Comment by Malabarba** [comment]
Done

**Comment by bbatsov** [comment]
:+1: Looks good! Guess now you can do the changes required for clojure-emacs/cider#1171 as well. :-)

**Comment by Malabarba** [comment]
Yep. I'll change the debugger to use these too instead of implementing its own.

**Comment by ryfow** [comment]
nice!

**Pull request merged**